lib/WorseReflection/Tests/Unit/Bridge/Phpactor/DocblockParser/PHPStanDocblockParserFactoryTest.php

```diff
@@ -0,0 +1,442 @@
+            new CallableType([
+                new StringType(),
+                new BooleanType(),
+            ], new StringType())
+        ];
+        yield [
+            '/** @return iterable */',
+            new PseudoIterableType(),
+        ];
+        yield [
+            '/** @return object */',
+            new ObjectType(),
+        ];
+        yield [
+            '/** @return resource */',
+            new ResourceType(),
+        ];
+
+        yield [
+            '/** @return void */',
+            new VoidType(),
+        ];
+        yield [
+            '/** @return array<int, string> */',
+            new ArrayType(new IntType(), new StringType())
+        ];
+        yield [
+            '/** @return array<int, array<string,bool>> */',
+            new ArrayType(new IntType(), new ArrayType(new StringType(), new BooleanType()))
+        ];
+
+        yield 'nullable' => [
+            '/** @return ?string */',
+            '?string',
+        ];
+
+        yield [
+            '/** @return T */',
+            'T',
+        ];
+
+        yield [
+            '/** @return \IteratorAggregate<Foobar> */',
+            'IteratorAggregate<Foobar>',
+        ];
+
+        yield [
+            '/** @return array{} */',
+            'array{}',
+        ];
+
+        yield 'arrayshape with keys' => [
+            '/** @return array{foo:int,bar:string} */',
+            'array{foo:int,bar:string}',
+        ];
+
+        yield 'parenthesized' => [
+            '/** @return null|(callable(int):string)|string|int */',
+            'null|(callable(int): string)|string|int',
+        ];
+
+        yield 'multiline array shape' => [
+            <<<'EOT'
+                /**
+                 * @return array{
+                 *   foo:int,
+                 *   bar:string
+                 * }
+                 */
+                EOT
+                ,
+            'array{foo:int,bar:string}',
+        ];
+
+        yield 'literals' => [
+            '/** @return null|"foo"|123|123.3 */',
+            'null|"foo"|123|123.3',
+        ];
+
+        yield 'list' => [
+            '/** @return list */',
+            TypeFactory::list(),
+        ];
+
+        yield 'list with type' => [
+            '/** @return list<string> */',
+            TypeFactory::list(TypeFactory::string()),
+        ];
+
+        yield 'never' => [
+            '/** @return never */',
+            TypeFactory::never(),
+        ];
+
+        yield 'false' => [
+            '/** @return false */',
+            TypeFactory::false(),
+        ];
+        yield 'union false' => [
+            '/** @return false|int */',
+            TypeFactory::union(TypeFactory::false(), TypeFactory::int())
+        ];
+
+        yield 'psalm prefix' => [
+            '/** @psalm-return int */',
+            TypeFactory::int(),
+        ];
+
+        yield 'conditional type' => [
+            '/** @return ($foo is true ? string : int) */',
+            TypeFactory::parenthesized(
```

Comment:
Again, PHPStan return it correctly, but without parenthesis

Reply:
ConditionalTypes are now wrapped in ParenthesizedType by default with PHPStan parser, but this seems fake to me